{
  int attempts=3;
  BackoffExponentiallyAndRetryOnThrowableCacheLoader<String,Boolean> backoff=new BackoffExponentiallyAndRetryOnThrowableCacheLoader<String,Boolean>(ResourceNotFoundException.class,50l,500l,attempts,mock);
  expect(mock.load("foo")).andThrow(new ResourceNotFoundException()).times(attempts);
  replay(mock);
  try {
    backoff.load("foo");
    assertTrue(false);
  }
 catch (  ResourceNotFoundException e) {
  }
  verify(mock);
}
